src/coprocessor/dag/rpn_expr/impl_arithmetic.rs

```diff
@@ -314,6 +314,74 @@ impl ArithmeticOp for DecimalMultiply {
     }
 }
 
+#[derive(Debug)]
+pub struct RealMultiply;
+
+impl ArithmeticOp for RealMultiply {
+    type T = Real;
+    fn calc(lhs: &Real, rhs: &Real) -> Result<Option<Real>> {
+        let res = *lhs * *rhs;
+        if res.is_infinite() {
+            Err(Error::overflow("REAL", &format!("({} * {})", lhs, rhs)).into())
+        } else {
+            Ok(Some(res))
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct IntIntMultiply;
+
+impl ArithmeticOp for IntIntMultiply {
+    type T = Int;
+    fn calc(lhs: &Int, rhs: &Int) -> Result<Option<Int>> {
+        lhs.checked_mul(*rhs)
+            .ok_or_else(|| Error::overflow("INT", &format!("({} * {})", lhs, rhs)).into())
```

Comment:
INT in MySQL is 4 bytes, and BIGINT is 8 bytes. So I think "BIGINT" may be more proper.